pkg/backup/item_backupper.go

```diff
@@ -253,11 +254,23 @@ func (ib *defaultItemBackupper) backupItem(logger logrus.FieldLogger, obj runtim
 		return false, kubeerrs.NewAggregate(backupErrs)
 	}
 
+	// group version of this object
+	// Used on filepath if backing up all groups and versions
+	version := resourceVersion(obj)
+
 	var filePath string
 	if namespace != "" {
-		filePath = filepath.Join(api.ResourcesDir, groupResource.String(), api.NamespaceScopedDir, namespace, name+".json")
+		if features.IsEnabled("Migration") {
```

Comment:
If the backup file format is backwards compatible on restore (meaning Velero v1.3.1 could restore a backup made with this code change), then I don't think the feature flag would be necessary anywhere in this PR.

Reply:
ok, sure. I will change the code accordingly and provide the testing of backward compatibility. And yes, I will provide documentation on the file format.

Reply:
Flagging as resolved.